### DISCUSSION TITLE
How do I get Emgu.CV.4.4.4099 with Cuda -  Emgu.CV.Util.CvException: 'OpenCV: The called functionality is disabled for current build or platform'

### DISCUSSION THREAD

**Discussion by NFran30** [emgucv/emgucv#408] [Q&A]
I have been trying to download Emgu.CV with Cuda. It's been a really interesting process and feel like I've jumped through circles but keep coming back to the same result. I would like to download the 4.4 version. I have tried downloading it from github and adding opencv and opencv_contrib to the folders in Emgu.CV to make sure that I can incorporate the open_cv portion with cuda. But when creating trying to build it in VS there was over 400 errors that came back, mostly having to deal with the Emgu.CV.Extern directory and VTK. So I after days of trying this method I decided to switch gears. 

I tried downloading it form sourceforce. After building by using Solution\Windows.Desktop I copied everything from libs/x64 to the debug folder of my current project. I've referenced Emgu.CV.Platform.NetStandard\AnyCPU\Debug\Emgu.CV.Platform.NetStandard. I have added items Emgu.CV.Runtime\Windows
\Emgu.CV.Runtime.Windows.shproj and Emgu.CV.Runtime.Windows.projitems into my project. I still can't figure out why the called functionality is disabled. Does anyone see what I may be doing wrong here?

**Comment by emgucv** [comment]
Seems like you have downloaded the version that is build without CUDA support. 

We just updated the release note to make it clear:
https://github.com/emgucv/emgucv/releases/tag/4.5.1 

> "libemgucv-windesktop-4.5.1.4349.exe" is an installer that will install Emgu CV (without CUDA support) into your workstation, it will also install debugger visualizer for visual studio 2019.
> 
> "libemgucv-windesktop-4.5.1.4349.zip" contains all files in the above installer, but will not install the debugger visualizer for visual studio.
> 
> "libemgucv-windesktop_x64-cuda-4.5.1.4349.zip.selfextract.exe" is a self extract zip file, with binary that contains "CUDA" support. It will require a NVidia graphic card with latest graphic card driver installed.
> 

If you need to use CUDA support, please download the file with "-cuda" in the file name.

**Reply by NFran30** [reply]
I have downloaded the Cuda version. I referenced the NetStandard .dll from the libs folder. I have also copied the x64 folder to the bin directory of my project. I have included the Emgu.CV.Runtime.Windows.shproj and referenced it in my project. 

![image](https://user-images.githubusercontent.com/78090775/107256107-1126c180-6a07-11eb-8436-0bca75778f4a.png)


I still get this error each time....
![emguerror](https://user-images.githubusercontent.com/78090775/107256016-f05e6c00-6a06-11eb-9149-ec0f3324094b.PNG)

**Reply by NFran30** [reply]
I have also updated my driver for me Nvidia Quadro RTX 4000, that didn't work either. If I call CudaInvoke.HasCuda it returns true. I am not sure what I am doing wrong here...I have tried all of the same steps with both emgucv/4.5.1/libemgucv-windesktop_x64-cuda-4.5.1.4349.zip.selfextract.exe and emgucv/4.4.0/libemgucv-windesktop_x64-cuda-4.4.0.4099.zip.selfextract.exe

**Reply by NFran30** [reply]
I have read some documentation that states that OpenCv doesn't support the CudaVideoReader in some of the newer Cuda toolkits. It looks for a cudacodec  library called NVCUVID  that isn't present in anything after Cuda 7.5. Has anyone else ran into this situation? If so do you know a work around?

**Reply by NFran30** [reply]
https://forums.developer.nvidia.com/t/opencv-error-the-function-feature-is-not-implemented-the-called-functionality-is-disabled-for-current-build-or-platform-in-throw-no-cuda/59361

**Comment by emgucv** [comment]
Open CV's implementation of CudaVideoReader is from old releases that are based on the NVCUVID library that has now been deprecated:
https://docs.nvidia.com/cuda/video-decoder/index.html

You can follow this Open CV tickets:
https://github.com/opencv/opencv/issues/11220

The comment from an Open CV developer mentions:
https://github.com/opencv/opencv/issues/11220#issuecomment-378582885

> OpenCV has no support for new API and there are no plans to add this.
> The latest CUDA version with NVCUVID is ~ CUDA 6.5.

The default Open CV cuda release no longer enable CudaVideoReader or CudaVideoWriter.

With the exception of CudaVideoReader and CudaVideoWriter, other CUDA functions should work if you have downloaded Emgu CV release with cuda.

**Reply by NFran30** [reply]
Ok thank you for getting back to me.
